ggdp/assets/allo.py

```diff
@@ -0,0 +1,219 @@
+import pandas as pd
+from dagster import Backoff, RetryPolicy, asset
+from regex import R
+
+from ..resources import GrantsStackIndexerGraphQL
+
+
+@asset
+def raw_allo_applications(indexer_graphql: GrantsStackIndexerGraphQL) -> pd.DataFrame:
+    query = """
+    {
+        applications {
+            anchorAddress
+            chainId
+            createdAtBlock
+            createdByAddress
+            id
+            metadata
+            metadataCid
+            nodeId
+            projectId
+            roundId
+            status
+            statusSnapshots
+            statusUpdatedAtBlock
+            tags
+            totalAmountDonatedInUsd
+            totalDonationsCount
+            uniqueDonorsCount
+        }
+    }
+    """
+    response = indexer_graphql.query(query)
+    df = pd.DataFrame(response["data"]["applications"]).convert_dtypes()
+    df["metadata"] = df["metadata"].apply(lambda x: json.dumps(x))
+
+    return df
+
+
+@asset
+def raw_allo_rounds(indexer_graphql: GrantsStackIndexerGraphQL) -> pd.DataFrame:
+    query = """
+    {
+        rounds {
+            adminRole
+            applicationMetadata
+            applicationMetadataCid
+            applicationsEndTime
+            applicationsStartTime
+            chainId
+            createdAtBlock
+            createdByAddress
+            donationsEndTime
+            donationsStartTime
+            id
+            isReadyForPayout
+            managerRole
+            matchAmount
+            matchAmountInUsd
+            matchTokenAddress
+            nodeId
+            projectId
+            roundMetadata
+            roundMetadataCid
+            strategyAddress
+            strategyId
+            strategyName
+            tags
+            totalAmountDonatedInUsd
+            totalDonationsCount
+            uniqueDonorsCount
+            updatedAtBlock
+        }
+    }
+    """
+    response = indexer_graphql.query(query)
+    df = pd.DataFrame(response["data"]["rounds"])
+
+    df["roundMetadata"] = df["roundMetadata"].apply(lambda x: json.dumps(x))
+    df["applicationMetadata"] = df["applicationMetadata"].apply(lambda x: json.dumps(x))
+    df["tags"] = df["tags"].apply(lambda x: json.dumps(x))
+
+    df = df.convert_dtypes()
+    return df
+
+
+@asset
+def raw_allo_donations(indexer_graphql: GrantsStackIndexerGraphQL) -> pd.DataFrame:
+    query = """
+        query($first: Int, $offset: Int) {
+            donations(first: $first, offset: $offset) {
+                amountInUsd
+                amount
+                amountInRoundMatchToken
+                applicationId
+                blockNumber
+                chainId
+                id
+                donorAddress
+                nodeId
+                projectId
+                recipientAddress
+                roundId
+                tokenAddress
+                transactionHash
+            }
+        }
+        """
+    response = indexer_graphql.paginated_query(query, size=50000)
```

Comment:
Page size seems to be a bit on larger side, but if API can handle that and so can GH runner, I have no strong objections.

Reply:
Made it 10.000! :see_no_evil: 

Still on the large size.